src/features/game/expansion/components/resources/Pebble.tsx

```diff
@@ -0,0 +1,254 @@
+  const [game] = useActor(gameService);
+
+  const [showPopover, setShowPopover] = useState(true);
+  const [popover, setPopover] = useState<JSX.Element | null>();
+
+  const [touchCount, setTouchCount] = useState(0);
+  // When to hide the pebble that pops out
+  const [collecting, setCollecting] = useState(false);
+
+  const containerRef = useRef<HTMLDivElement>(null);
+  const sparkGif = useRef<SpriteSheetInstance>();
+  const minedGif = useRef<SpriteSheetInstance>();
+
+  const [showPebbleTimeLeft, setShowPebbleTimeLeft] = useState(false);
+
+  const pebble = game.context.state.pebbles[pebbleIndex];
+  // Users will need to refresh to chop the tree again
+  const mined = !canMine(pebble);
+  const { setToast } = useContext(ToastContext);
+
+  // Reset the shake count when clicking outside of the component
+  useEffect(() => {
+    const handleClickOutside = (event: any) => {
+      if (
+        containerRef.current &&
+        !containerRef.current.contains(event.target)
+      ) {
+        setTouchCount(0);
+      }
+    };
+    document.addEventListener("click", handleClickOutside, true);
+    return () => {
+      document.removeEventListener("click", handleClickOutside, true);
+    };
+  }, []);
+
+  const displayPopover = async (element: JSX.Element) => {
+    setPopover(element);
+    setShowPopover(true);
+
+    await new Promise((resolve) => setTimeout(resolve, POPOVER_TIME_MS));
+    setShowPopover(false);
+  };
+
+  // Show/Hide Time left on hover
+
+  const handleMouseHoverPebble = () => {
+    if (mined) setShowPebbleTimeLeft(true);
+  };
+
+  const handleMouseLeavePebble = () => {
+    setShowPebbleTimeLeft(false);
+  };
+
+  const strike = () => {
+    const isPlaying = sparkGif.current?.getInfo("isPlaying");
+
+    if (!isPlaying) {
+      miningAudio.play();
+
+      sparkGif.current?.goToAndPlay(0);
+
+      setTouchCount((count) => count + 1);
+
+      // On second strike, mine
+      if (touchCount > 0 && touchCount === HITS - 1) {
+        mine();
+        miningFallAudio.play();
+        setTouchCount(0);
+      }
+    } else return;
+  };
+
+  const mine = async () => {
+    setTouchCount(0);
+
+    try {
+      gameService.send("pebble.struck", {
+        index: pebbleIndex,
+      });
+      setCollecting(true);
+      minedGif.current?.goToAndPlay(0);
+
+      displayPopover(
+        <div className="flex">
+          <img src={smallStone} className="w-5 h-5 mr-2" />
+          <span className="text-sm text-white text-shadow">{`+${pebble.stone.amount}`}</span>
+        </div>
+      );
+
+      setToast({
+        icon: smallStone,
+        content: `+${pebble.stone.amount}`,
+      });
+
+      await new Promise((res) => setTimeout(res, 2000));
+      setCollecting(false);
+    } catch (e: any) {
+      displayPopover(
+        <span className="text-xs text-white text-shadow">{e.message}</span>
+      );
+    }
+  };
+
+  const timeLeft = getTimeLeft(pebble.stone.minedAt, PEBBLE_RECOVERY_TIME);
```

Comment:
<img width="679" alt="Screen Shot 2022-07-08 at 9 25 43 am" src="https://user-images.githubusercontent.com/11745561/177887671-8ca7314c-6e8f-4d5b-8194-119559bfd99e.png">

It looks like the API payload does not include stone. This is causing a white screen. I would suggest updating the types to make `stone` optional and then checking whether it exists here.

<img width="582" alt="Screen Shot 2022-07-08 at 9 26 16 am" src="https://user-images.githubusercontent.com/11745561/177887690-0cb7eecd-48b3-43c5-ab54-c14d2baab1e4.png">

Reply:
This API payload is not correct due to you having an old record `pebbles` in your gameState in testnet. This was the shape of the payload in the past iteration. You will need to delete your current `pebbles` and then reload to get a clean one. 

The `undefined` issue should still be handled though. `stone` should definitely be optional. It is on the api side.